### PR TITLE
Added basic upgrading instructions

### LINKED ISSUE #5155: Document how to upgrade vue-cli
### What problem does this feature solve?
There's currently no documented way to upgrade vue-cli, however there is `vue upgrade --all`. If you search current documentation for 'upgrade', you find only one mention for the word and even that is just marketing speak. 

We should have a documented way to do upgrades to `vue-cli` inside the repo so when for example user sees a security issue in Vue UI dashboard in vue-cli packages, there would be a documented way to do the upgrade.

### What does the proposed API look like?
Start with minimal and easiest documentation, expand later (when we for example have non-experimental command available). I'm happy to start a PR if it helps.

### PR DESCRIPTION
**Pull request by Uninen**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [x] Docs
- [ ] Underlying tools
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**Other information:**

Fixes #5155

### CODE REVIEW

**Comment by Uninen** [comment]
The first stab at adding upgrade instructions adds two parts 1) upgrading your global packages, and 2) upgrading packages inside your project. This distinction seems to be not very clear to all new users.

What I'd like to add also, but did not add yet, is an additional comment about upgrades that break (which again seems to be a somewhat usual but many times easily fixable situation) with something in the lines of:

"If there is an error with the upgrade, a clean install by manually removing and then installing may resolve the issue.

``` bash
npm uninstall -g @vue/cli
npm install -g @vue/cli
# OR
yarn global remove @vue/cli
yarn global add @vue/cli
```
"

File: docs/guide/installation.md [hunk 1]
```diff
@@ -24,3 +24,32 @@ You can check you have the right version with this command:
 ```bash
 vue --version
 ```
+
+### Upgrading
+
+To upgrade the global Vue CLI package run:
```
Comment (NataliaTepluhina) [inline docs/guide/installation.md]:
```suggestion
To upgrade the global Vue CLI package, you need to run:
```

File: docs/guide/installation.md [hunk 2]
```diff
@@ -24,3 +24,32 @@ You can check you have the right version with this command:
...
+``` bash
+npm update -g @vue/cli
+
+# OR
+yarn global upgrade --latest @vue/cli
+```
+
+#### Project Dependencies
+
+Above upgrade commands apply to the global Vue CLI installation. To upgrade one or more `@vue/cli` related packages (including packages starting with `@vue/cli-plugin-`) inside your project, run `vue upgrade` inside the project directory:
```
Comment (NataliaTepluhina) [inline docs/guide/installation.md]:
```suggestion
Upgrade commands shown above apply to the global Vue CLI installation. To upgrade one or more `@vue/cli` related packages (including packages starting with `@vue/cli-plugin-`) inside your project, run `vue upgrade` inside the project directory:
```

**Comment by NataliaTepluhina** [comment]
Thanks for the changes!

**Pull request merged**